pkg/inference/scheduling/loader.go

```diff
@@ -372,9 +372,15 @@ func (l *loader) load(ctx context.Context, backendName, model string, mode infer
 		// See if we can satisfy the request with an existing runner.
 		existing, ok := l.runners[runnerKey{backendName, model, mode}]
 		if ok {
-			l.references[existing] += 1
-			l.timestamps[existing] = time.Time{}
-			return l.slots[existing], nil
+			select {
+			case <-l.slots[existing].done:
```

Comment:
I think it'd make sense to also run `l.evictRunner(backendName, model`) so we don't have to evict all runners in order to find a free slot. WDYT?

Reply:
Yep, that makes sense.